lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/SecurityTokenHelper.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.bridge.client;
+
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.security.token.Token;
+import javax.xml.bind.DatatypeConverter;
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Helper class to provide security token methods.
+ */
+final class SecurityTokenHelper {
+
+  private static final Logger LOG = Logger.getLogger(SecurityTokenHelper.class.getName());
+  private static final String TOKEN_PART_SEPARATOR = ",";
+  static final String TOKEN_KIND_SEPARATOR = ":";
+
+  private SecurityTokenHelper() {
+  }
+
+  /**
+   * Get list of TokenInfo.
+   *
+   * @param securityTokenIdentifierFile - the file that contains one or more security token ids.
+   * @param securityTokenPasswordFile - the file that contains one or more security token passwords.
+   * @param tokenKindString - array of strings that contains token kind strings.
+   * @return
+   * @throws IOException
+   */
+  static List<TokenInfo> getTokenInfo(final String securityTokenIdentifierFile,
+                                      final String securityTokenPasswordFile,
+                                      final String[] tokenKindString) throws IOException {
+
+    final List<TokenInfo> tokens = new ArrayList<>();
+    for (final String tokenStr : tokenKindString) {
+      LOG.log(Level.FINE, "tokenKind for each token:{0}.", tokenStr);
+      final String[] tokenParts = tokenStr.split(TOKEN_PART_SEPARATOR);
+      if (tokenParts.length != 2) {
+        LOG.log(Level.SEVERE, "Expected token parts length is 2 with format [tokenKind, tokenLen].");
+        throw new IllegalArgumentException("Expected token parts length is 2");
+      }
+
+      tokens.add(new TokenInfo(tokenParts[0], tokenParts[0], Integer.parseInt(tokenParts[1])));
+    }
+
+    try (final FileInputStream fStream = new FileInputStream(securityTokenPasswordFile)) {
+      try (final BufferedReader reader = new BufferedReader(new InputStreamReader(fStream, "UTF-8"))) {
```

Comment:
no need to have two embedded `try` blocks - simply do
```java
try (final BufferedReader reader = new BufferedReader(
    new InputStreamReader(new FileInputStream(securityTokenPasswordFile), "UTF-8"))) { // ...
```

Reply:
Good point